pfioh/swift_store.py

```diff
@@ -0,0 +1,266 @@
+"""
+Handle Swift File Storage Option
+"""
+
+import ast
+import base64
+import datetime
+import zlib
+import zipfile
+import os
+import configparser
+from pfioh import StoreHandler
+from pfioh import base64_process, zip_process, zipdir
+from keystoneauth1.identity import v3
+from keystoneauth1 import session
+from swiftclient import client as swift_client
+try:
+    from ._colors import Colors
+except:
+    from _colors import Colors
+
+class SwiftStore(StoreHandler):
+
+    swiftConnection = ''
+    
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.qprint('SwiftStore initialized')
+        
+
+    def _getScopedSession(self, osAuthUrl, username, password, osProjectDomain, osProjectName):
+        """
+        Uses keystone authentication to create and return a scoped session
+        """
+
+        passwordAuth  = v3.Password(auth_url= osAuthUrl,
+                            user_domain_name= 'default',
+                            username= username, password= password,
+                            project_domain_name= osProjectDomain,
+                            project_name= osProjectName,
+                            unscoped= False)
+
+        scopedSession = session.Session(auth= passwordAuth)
+        return scopedSession
+
+
+    def _initiateSwiftConnection(self, **kwargs):
+        """
+        Initiates a Swift connection and returns a Swift Connection Object
+        Swift credentials should be stored as a cfg file at /etc/swift 
+        """
+
+        for k,v in kwargs:
+            if k == 'configPath': configPath= v
+
+        configPath= '/etc/swift/swift-credentials.cfg'
+
+        config= configparser.ConfigParser()
+        f = open(configPath, 'r')
+        config.readfp(f)
+        f.close()
+        
+        osAuthUrl = config['AUTHORIZATION']['osAuthUrl']
+        username= config['AUTHORIZATION']['username']
+        password= config['AUTHORIZATION']['password']
+        osProjectDomain= config['PROJECT']['osProjectDomain']
+        osProjectName= config['PROJECT']['osProjectName']
+        
+        scopedSession = self._getScopedSession(osAuthUrl, username, password, osProjectDomain, osProjectName)
+        self.swiftConnection = swift_client.Connection(session=scopedSession)
+            
+
+    def _deleteEmptyDirectory(self, key):
+        """
+        Deletes the empty directory created by Swift in the parent directory
+        """
+
+        directoryPath = os.path.join(os.path.dirname(__file__), '../%s'%key)
+        try:
+            os.rmdir(directoryPath)
+            self.qprint("Temporary directory %s deleted"%key)
+        except:
+            self.qprint("No temporary directory found")
+
+
+    def _putContainer(self, key):
+        """
+        Creates a container with the name as the key
+        """
+
+        try:
+            self.swiftConnection.put_container(key)
+            self.qprint('Container added sucessfully')

```

Comment:
successfully

Reply:
Also maybe change the comment to swift container added successfully to make it clear what kind of container you are talking about.